forc-plugins/forc-crypto/src/keys/vanity.rs

```diff
@@ -0,0 +1,551 @@
+
+    /// Timeout in seconds for address generation
+    #[arg(long, value_name = "SECONDS")]
+    pub timeout: Option<u64>,
+
+    /// Return mnemonic with address (default false)
+    #[arg(long)]
+    pub mnemonic: bool,
+
+    /// Path to save the generated vanity address to.
+    #[arg(long, value_hint = clap::ValueHint::FilePath, value_name = "PATH")]
+    pub save_path: Option<PathBuf>,
+}
+
+pub fn handler(args: Arg) -> anyhow::Result<serde_json::Value> {
+    let Arg {
+        starts_with,
+        ends_with,
+        mnemonic,
+        save_path,
+        timeout,
+    } = args;
+
+    let mut left_exact_hex = None;
+    let mut left_regex = None;
+    if let Some(prefix) = starts_with {
+        match parse_pattern(&prefix, true)? {
+            Either::Left(left) => left_exact_hex = Some(left),
+            Either::Right(re) => left_regex = Some(re),
+        }
+    }
+
+    let mut right_exact_hex = None;
+    let mut right_regex = None;
+    if let Some(suffix) = ends_with {
+        match parse_pattern(&suffix, false)? {
+            Either::Left(right) => right_exact_hex = Some(right),
+            Either::Right(re) => right_regex = Some(re),
+        }
+    }
+
+    println!("Starting to generate vanity address...");
+    let start_time = Instant::now();
+
+    let result = match (left_exact_hex, left_regex, right_exact_hex, right_regex) {
+        (Some(left), _, Some(right), _) => {
+            let matcher = HexMatcher { left, right };
+            find_vanity_address_with_timeout(matcher, mnemonic, timeout)
+        }
+        (Some(left), _, _, Some(right)) => {
+            let matcher = LeftExactRightRegexMatcher { left, right };
+            find_vanity_address_with_timeout(matcher, mnemonic, timeout)
+        }
+        (_, Some(left), _, Some(right)) => {
+            let matcher = RegexMatcher { left, right };
+            find_vanity_address_with_timeout(matcher, mnemonic, timeout)
+        }
+        (_, Some(left), Some(right), _) => {
+            let matcher = LeftRegexRightExactMatcher { left, right };
+            find_vanity_address_with_timeout(matcher, mnemonic, timeout)
+        }
+        (Some(left), None, None, None) => {
+            let matcher = LeftHexMatcher { left };
+            find_vanity_address_with_timeout(matcher, mnemonic, timeout)
+        }
+        (None, None, Some(right), None) => {
+            let matcher = RightHexMatcher { right };
+            find_vanity_address_with_timeout(matcher, mnemonic, timeout)
+        }
+        (None, Some(re), None, None) => {
+            let matcher = SingleRegexMatcher { re };
+            find_vanity_address_with_timeout(matcher, mnemonic, timeout)
+        }
+        (None, None, None, Some(re)) => {
+            let matcher = SingleRegexMatcher { re };
+            find_vanity_address_with_timeout(matcher, mnemonic, timeout)
+        }
+        _ => {
+            return Err(VanityAddressError::InvalidPattern(
```

Comment:
nit: I think this should be a separate error, `InvalidPatternCombination`, since it is different and it will make the error output a little cleaner.

Reply:
somewhat addressed by the refactor: https://github.com/FuelLabs/sway/pull/6661/commits/59ab719069be67c3a2ef209e632c2d5039996746

Reply:
removed custom error as no longer needed as it did not see much usage; reverting back to anyhow: https://github.com/FuelLabs/sway/pull/6661/commits/80ca7a2ce4a87e0c8a6038ff69defc3e07af117d